src/network/networkd-wifi.c

```diff
@@ -12,6 +14,68 @@
 #include "string-util.h"
 #include "wifi-util.h"
 
+int link_rfkilled(Link *link) {
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        const char *s;
+        int r;
+
+        assert(link);
+
+        if (link->iftype != ARPHRD_ETHER)
+                return false;
+
+        if (!link->sd_device)
+                return false;
+
+        r = sd_device_get_devtype(link->sd_device, &s);
+        if (r < 0)
+                return r;
+
+        if (!streq_ptr(s, "wlan"))
+                return false;
+
+        r = sd_device_get_syspath(link->sd_device, &s);
+        if (r < 0)
+                return r;
+
+        s = strjoina(s, "/phy80211/rfkill1");
```

Comment:
i think you need to search for the name here. i.e. iterate through all subdirs of the phy80211/ dir, then check every single one starting with "rfkill" in the name. I think there might be multiple rfkill devices associated with one phy80211 dir